subsys/bluetooth/host/adv.c

```diff
@@ -484,39 +524,43 @@ static int hci_set_adv_ext_fragmented(struct bt_le_ext_adv *adv, uint16_t hci_op
 						/* We can always set the data in two operations
 						 * Therefore, we know that this is the last.
 						 */
-						set_data->op = BT_HCI_LE_EXT_ADV_OP_LAST_FRAG;
+						set_data[op_offset] =
+							BT_HCI_LE_EXT_ADV_OP_LAST_FRAG;
 					} else if (len + 2 <= BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN) {
 						/* First part fits. */
-						set_data->op = BT_HCI_LE_EXT_ADV_OP_LAST_FRAG;
+						set_data[op_offset] =
+							BT_HCI_LE_EXT_ADV_OP_LAST_FRAG;
 					} else {
 						/* The data must be split into two
 						 * commands.
 						 */
-						set_data->op = BT_HCI_LE_EXT_ADV_OP_INTERM_FRAG;
+						set_data[op_offset] =
+							BT_HCI_LE_EXT_ADV_OP_INTERM_FRAG;
 					}
 				} else {
-					set_data->op = BT_HCI_LE_EXT_ADV_OP_INTERM_FRAG;
+					set_data[op_offset] = BT_HCI_LE_EXT_ADV_OP_INTERM_FRAG;
 				}
 
 				if (offset == 0) {
-					set_data->len = MIN(len + 2,
-							    BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN);
+					set_data[len_offset] =
+						MIN(len + 2, BT_HCI_LE_EXT_ADV_FRAG_MAX_LEN);
 				} else {
 					/* No need to take min operation here,
 					 * as we can always fit the second part.
 					 */
-					set_data->len = len - offset;
+					set_data[len_offset] = len - offset;
 				}
 
 				if (offset == 0) {
-					set_data->data[0] = len + 1;
-					set_data->data[1] = type;
-					memcpy(&set_data->data[2], data[j].data, set_data->len - 2);
-					offset += set_data->len - 2;
+					set_data[data_offset + 0] = len + 1;
+					set_data[data_offset + 1] = type;
+					memcpy(&set_data[data_offset + 2], data[j].data,
+					       set_data[len_offset] - 2);
```

Comment:
Given that you already started using `_offset` variables, consider adding variables `data_len_offset`, `data_type_offset` and `data_value_offset`